projects/components/src/filter-button/in-filter-button.component.ts

```diff
@@ -0,0 +1,97 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { TypedSimpleChanges } from '@hypertrace/common';
+import { FilterAttribute } from '../filter-bar/filter-attribute';
+import { UserFilterOperator } from '../filter-bar/filter/filter-api';
+import { IconSize } from '../icon/icon-size';
+import { FilterButtonService } from './filter-button.service';
+
+@Component({
+  selector: 'htc-in-filter-button',
+  styleUrls: ['./in-filter-button.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="filter-button" *ngIf="this.isSupported">
+      <htc-popover [closeOnClick]="false" (popoverOpen)="this.onPopoverOpen()" (popoverClose)="this.onPopoverClose()">
+        <htc-popover-trigger>
+          <div #trigger>
+            <htc-icon icon="${IconType.Filter}" size="${IconSize.Small}"></htc-icon>
+          </div>
+        </htc-popover-trigger>
+        <htc-popover-content>
+          <div [style.min-width.px]="trigger.offsetWidth" class="popover-content">
+            <div *ngFor="let availableValue of this.values" class="popover-item">
+              <htc-checkbox
+                [label]="availableValue"
+                [checked]="this.selected.has(availableValue)"
+                (checkedChange)="this.onChecked($event, availableValue)"
+              ></htc-checkbox>
+            </div>
+          </div>
+        </htc-popover-content>
+      </htc-popover>
+    </div>
+  `
+})
+export class InFilterButtonComponent<T = unknown> implements OnChanges {
+  @Input()
+  public metadata?: FilterAttribute[];
+
+  @Input()
+  public attribute?: FilterAttribute;
+
+  @Input()
+  public values?: T[];
+
+  @Output()
+  public readonly popoverOpen: EventEmitter<boolean> = new EventEmitter();
+
+  public isSupported: boolean = false;
+  public selected: Set<T> = new Set<T>();
+
+  public constructor(private readonly filterButtonService: FilterButtonService) {}
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.attribute) {
+      this.isSupported = this.attribute
+        ? this.filterButtonService.isSupportedOperator(this.attribute, UserFilterOperator.In)
```

Comment:
I wish there was a way to remove these codecov messages. :P 
Makes it hard to read

Reply:
Agreed!